src/decode.c

```diff
@@ -74,6 +74,31 @@ extern bool stats_decoder_events;
 extern const char *stats_decoder_events_prefix;
 extern bool stats_stream_events;
 uint8_t decoder_max_layers = PKT_DEFAULT_MAX_DECODED_LAYERS;
+uint16_t packet_alert_max = PACKET_ALERT_MAX;
+
+/**
+ * \brief Initialize PacketAlerts with dynamic alerts array size
+ *
+ */
+PacketAlert *CreatePacketAlert(void)
+{
+    PacketAlert *pa_array = NULL;
+    if (RunmodeIsUnittests()) {
+        pa_array = SCCalloc(PACKET_ALERT_MAX, sizeof(PacketAlert));
```

Comment:
Do we have any idea what is the performance impact ?
Could/should we use a pool of preallocated buffers ?

Reply:
The packets themselves are pooled and reused, so this allocation is a once per packet step.